src/test/java/com/datastax/mcac/BaseIntegrationTest.java

```diff
@@ -30,9 +30,17 @@
     protected final String version;
     protected static DockerHelper docker;
 
-    @Parameterized.Parameters
+    @Parameterized.Parameters(name = "{0}")
     public static Iterable<String[]> functions()
     {
+        String cassandraVersion = System.getProperty("cassandra.version");
+        if (cassandraVersion != null && cassandraVersion.startsWith("4.1"))
+        {
+            String[] versions = new String[]{"4.1"};
+            ArrayList<String[]> list = new ArrayList<>(1);
+            list.add(versions);
+            return list;
+        }
```

Comment:
This alters the test matrix based on what version of Cassandra libs we are building and testing against. Tests for Cassandra 4.0.x and lower won't work if we build against 4.1, and tests for 4.1 won't work if we build against older Cassandra libs.